src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -1893,14 +1903,15 @@ public StringBuilder Replace(string oldValue, string newValue, int startIndex, i
 
             int deltaLength = newValue.Length - oldValue.Length;
 
-            int[] replacements = null; // A list of replacement positions in a chunk to apply
+            int[]? replacements = null; // A list of replacement positions in a chunk to apply
```

Comment:
Can't comment on the line directly, but Replace allows `newValue` to be null.  Seems like it should be `string?`, both in this overload and the other one?

Reply:
good catch, I've changed this and the other `Replace`